src/users/users.ts

```diff
@@ -0,0 +1,135 @@
+import { WorkOS } from '../workos';
+import {
+  AddUserToOrganizationOptions,
+  AuthenticateManagedUserOptions,
+  AuthenticateUnmanagedUserOptions,
+  AuthenticationResponse,
+  CompletePasswordResetOptions,
+  CreateEmailVerificationChallengeOptions,
+  CreateEmailVerificationChallengeResponse,
+  CreatePasswordResetChallengeOptions,
+  CreatePasswordResetChallengeResponse,
+  CreateUnmanagedUserOptions,
+  ListUsersOptions,
+  RemoveUserFromOrganizationOptions,
+  RevokeSessionOptions,
+  User,
+  VerifySessionOptions,
+  VerifySessionResponse,
+} from './interfaces';
+import { List } from '../common/interfaces/list.interface';
+
+export class Users {
+  constructor(private readonly workos: WorkOS) {}
+
+  async getUser(userId: string): Promise<User> {
+    const { data } = await this.workos.get(`/users/${userId}`);
+    return data;
+  }
+
+  async listUsers(options?: ListUsersOptions): Promise<List<User>> {
+    const { data } = await this.workos.get('/users', {
+      query: options,
+    });
+    return data;
+  }
+
+  async createUnmanagedUser(
+    payload: CreateUnmanagedUserOptions,
+  ): Promise<User> {
+    const { data } = await this.workos.post('/users', payload);
+
+    return data;
+  }
+
+  async authenticateUnmanagedUser(
+    payload: AuthenticateUnmanagedUserOptions,
+  ): Promise<AuthenticationResponse> {
+    const { data } = await this.workos.post('/users/authentications', payload);
+    return data;
+  }
+
+  async authenticateManagedUser(
```

Comment:
Similar to the password one, the authentication mechanism is actually what separate this endpoint apart, not the type of user it returns.

```suggestion
  async authenticateUserWithToken(
```

Reply:
done